In [4]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as pyplot
import os 
import glob
import pickle
from tqdm import tqdm

In [1]:
result_directory = r"D:\Road to PhD\Data Base\bit_frequency_results"
print("Files in the result directory:")
print(os.listdir(result_directory))

Files in the result directory:
['chunk_0_100_simplified_bit_frequency.pkl', 'chunk_1000_1100_simplified_bit_frequency.pkl', 'chunk_100_200_simplified_bit_frequency.pkl', 'chunk_1100_1200_simplified_bit_frequency.pkl', 'chunk_1200_1300_simplified_bit_frequency.pkl', 'chunk_1300_1400_simplified_bit_frequency.pkl', 'chunk_1400_1500_simplified_bit_frequency.pkl', 'chunk_1500_1600_simplified_bit_frequency.pkl', 'chunk_1600_1700_simplified_bit_frequency.pkl', 'chunk_1700_1800_simplified_bit_frequency.pkl', 'chunk_1800_1900_simplified_bit_frequency.pkl', 'chunk_1900_2000_simplified_bit_frequency.pkl', 'chunk_2000_2100_simplified_bit_frequency.pkl', 'chunk_200_300_simplified_bit_frequency.pkl', 'chunk_2100_2200_simplified_bit_frequency.pkl', 'chunk_2200_2300_simplified_bit_frequency.pkl', 'chunk_2300_2400_simplified_bit_frequency.pkl', 'chunk_2400_2417_simplified_bit_frequency.pkl', 'chunk_300_400_simplified_bit_frequency.pkl', 'chunk_400_500_simplified_bit_frequency.pkl', 'chunk_500_600_simpl

In [4]:
file_path = r"D:\Road to PhD\Data Base\bit_frequency_results\chunk_0_100_simplified_bit_frequency.pkl"
try:
    specific_chunk = pd.read_pickle(file_path)
    print(specific_chunk.info())  # Structure of the DataFrame
    print(specific_chunk.head())  # Preview the data
except Exception as e:
    print(f"Error loading file: {e}")

<class 'pandas.core.frame.DataFrame'>
Index: 2048 entries, 597 to 72
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Bit (+1)          2048 non-null   object
 1   Value (+1)        2048 non-null   int64 
 2   Frequency (+1)    2048 non-null   int64 
 3   Molecule_ID (+1)  2048 non-null   object
 4   Bit (-1)          2048 non-null   object
 5   Value (-1)        2048 non-null   int64 
 6   Frequency (-1)    2048 non-null   int64 
 7   Molecule_ID (-1)  2048 non-null   object
 8   Bit (0)           2048 non-null   object
 9   Value (0)         2048 non-null   int64 
 10  Frequency (0)     2048 non-null   int64 
 11  Molecule_ID (0)   2048 non-null   object
dtypes: int64(6), object(6)
memory usage: 208.0+ KB
None
     Bit (+1)  Value (+1)  Frequency (+1)  \
597       598           1           17472   
427       428           1           17050   
1555     1556           1           16520   
1989     1990    

In [26]:
specific_chunk.head()


Bit (+1)  Value (+1)  Frequency (+1)  \
597       598           1           17472   
427       428           1           17050   
1555     1556           1           16520   
1989     1990           1           14553   
7           8           1           14392   

                                       Molecule_ID (+1) Bit (-1)  Value (-1)  \
597   ((43, 1448), (43, 1541), (114, 1448), (114, 15...      598          -1   
427   ((118, 5750), (119, 5750)), ((94, 1894),), ((4...      428          -1   
1555  ((88, 1693), (88, 2517), (88, 4942), (88, 5508...     1556          -1   
1989  ((74, 3532),), ((43, 1448), (43, 1541), (114, ...     1990          -1   
7     ((230, 1589), (230, 3933)), ((197, 4987), (198...        8          -1   

      Frequency (-1)                                   Molecule_ID (-1)  \
597             9141  ((139, 422),), ((140, 2976), (140, 3690), (141...   
427             6292  ((76, 3223),), ((161, 2673),), ((192, 1641), (...   
1555            9800  ((67, 3486), (68, 3486), (69, 3486)), ((76, 32...   
1989           10296  ((151, 227), (151, 1638), (151, 3456), (152, 2...   
7               8284  ((139, 422),), ((140, 2976), (140, 3690), (141...   

     Bit (0)  Value (0)  Frequency (0)  \
597      598          0          25332   
427      428          0          28603   
1555    1556          0          25625   
1989    1990          0          27096   
7          8          0          29269   

                                        Molecule_ID (0)  
597   ((74, 3532),), ((219, 3141), (219, 4381)), ((0...  
427   ((139, 422),), ((140, 2976), (140, 3690), (141...  
1555  ((139, 422),), ((140, 2976), (140, 3690), (141...  
1989  ((139, 422),), ((140, 2976), (140, 3690), (141...  
7     ((67, 3486), (68, 3486), (69, 3486)), ((76, 32...

In [27]:
# Directory where results are saved
result_directory = r"D:\Road to PhD\Data Base\bit_frequency_results"

# Initialize empty DataFrames for each frequency category
results_frequency_plus1 = pd.DataFrame()
results_frequency_minus1 = pd.DataFrame()
results_frequency_0 = pd.DataFrame()

# Process each file incrementally
for file in tqdm(os.listdir(result_directory), desc="Processing files incrementally"):
    if file.endswith(".pkl"):  # Ensure only .pkl files are processed
        chunk_name = file.replace(".pkl", "")  # Extract chunk name
        file_path = os.path.join(result_directory, file)

        try:
            # Load the file
            specific_chunk = pd.read_pickle(file_path)
            print(f"Processing {chunk_name} with shape: {specific_chunk.shape}")

            # Aggregate and rank for each frequency category separately
            # +1 Frequency
            plus1_summary = specific_chunk.groupby('Bit (+1)').agg({'Frequency (+1)': 'sum'}).reset_index()
            plus1_summary['Chunk_Name'] = chunk_name  # Add source chunk info
            results_frequency_plus1 = pd.concat([results_frequency_plus1, plus1_summary], ignore_index=True)

            # -1 Frequency
            minus1_summary = specific_chunk.groupby('Bit (-1)').agg({'Frequency (-1)': 'sum'}).reset_index()
            minus1_summary['Chunk_Name'] = chunk_name  # Add source chunk info
            results_frequency_minus1 = pd.concat([results_frequency_minus1, minus1_summary], ignore_index=True)

            # 0 Frequency
            zero_summary = specific_chunk.groupby('Bit (0)').agg({'Frequency (0)': 'sum'}).reset_index()
            zero_summary['Chunk_Name'] = chunk_name  # Add source chunk info
            results_frequency_0 = pd.concat([results_frequency_0, zero_summary], ignore_index=True)

        except Exception as e:
            print(f"Error processing file {file}: {e}")

# Combine aggregated results and rank them for each frequency category
final_plus1 = results_frequency_plus1.groupby('Bit (+1)').agg({'Frequency (+1)': 'sum'}).reset_index()
final_minus1 = results_frequency_minus1.groupby('Bit (-1)').agg({'Frequency (-1)': 'sum'}).reset_index()
final_0 = results_frequency_0.groupby('Bit (0)').agg({'Frequency (0)': 'sum'}).reset_index()

# Sort each final result by its frequency in descending order
final_plus1 = final_plus1.sort_values(by='Frequency (+1)', ascending=False)
final_minus1 = final_minus1.sort_values(by='Frequency (-1)', ascending=False)
final_0 = final_0.sort_values(by='Frequency (0)', ascending=False)

# Save the final ranked results to disk
output_directory = r"D:\Road to PhD\Data Base\ranked_frequency_results"
os.makedirs(output_directory, exist_ok=True)  # Create the directory if it doesn't exist

final_plus1.to_pickle(os.path.join(output_directory, "final_ranked_frequency_plus1.pkl"))
final_minus1.to_pickle(os.path.join(output_directory, "final_ranked_frequency_minus1.pkl"))
final_0.to_pickle(os.path.join(output_directory, "final_ranked_frequency_0.pkl"))

print("Final ranked results saved successfully!")

Processing files incrementally:   4%|█▉                                                 | 1/26 [00:02<00:52,  2.09s/it]

Processing chunk_0_100_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:   8%|███▉                                               | 2/26 [00:06<01:18,  3.27s/it]

Processing chunk_1000_1100_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  12%|█████▉                                             | 3/26 [00:25<04:04, 10.61s/it]

Processing chunk_100_200_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  15%|███████▊                                           | 4/26 [00:46<05:20, 14.58s/it]

Processing chunk_1100_1200_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  19%|█████████▊                                         | 5/26 [01:07<05:53, 16.84s/it]

Processing chunk_1200_1300_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  23%|███████████▊                                       | 6/26 [01:27<06:03, 18.19s/it]

Processing chunk_1300_1400_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  27%|█████████████▋                                     | 7/26 [01:48<06:03, 19.11s/it]

Processing chunk_1400_1500_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  31%|███████████████▋                                   | 8/26 [02:09<05:52, 19.58s/it]

Processing chunk_1500_1600_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  35%|█████████████████▋                                 | 9/26 [02:29<05:37, 19.84s/it]

Processing chunk_1600_1700_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  38%|███████████████████▏                              | 10/26 [02:50<05:23, 20.22s/it]

Processing chunk_1700_1800_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  42%|█████████████████████▏                            | 11/26 [03:11<05:07, 20.47s/it]

Processing chunk_1800_1900_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  46%|███████████████████████                           | 12/26 [03:33<04:49, 20.67s/it]

Processing chunk_1900_2000_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  50%|█████████████████████████                         | 13/26 [03:54<04:30, 20.77s/it]

Processing chunk_2000_2100_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  54%|██████████████████████████▉                       | 14/26 [04:13<04:05, 20.44s/it]

Processing chunk_200_300_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  58%|████████████████████████████▊                     | 15/26 [04:34<03:46, 20.62s/it]

Processing chunk_2100_2200_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  62%|██████████████████████████████▊                   | 16/26 [04:55<03:26, 20.69s/it]

Processing chunk_2200_2300_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  65%|████████████████████████████████▋                 | 17/26 [05:16<03:06, 20.75s/it]

Processing chunk_2300_2400_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  69%|██████████████████████████████████▌               | 18/26 [05:20<02:05, 15.68s/it]

Processing chunk_2400_2417_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  73%|████████████████████████████████████▌             | 19/26 [05:40<01:59, 17.08s/it]

Processing chunk_300_400_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  77%|██████████████████████████████████████▍           | 20/26 [06:02<01:51, 18.61s/it]

Processing chunk_400_500_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  81%|████████████████████████████████████████▍         | 21/26 [06:24<01:38, 19.63s/it]

Processing chunk_500_600_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  85%|██████████████████████████████████████████▎       | 22/26 [06:47<01:21, 20.37s/it]

Processing chunk_600_700_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  88%|████████████████████████████████████████████▏     | 23/26 [07:09<01:02, 20.86s/it]

Processing chunk_700_800_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  92%|██████████████████████████████████████████████▏   | 24/26 [07:30<00:42, 21.09s/it]

Processing chunk_800_900_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally:  96%|████████████████████████████████████████████████  | 25/26 [07:52<00:21, 21.32s/it]

Processing chunk_900_1000_simplified_bit_frequency with shape: (2048, 12)


Processing files incrementally: 100%|██████████████████████████████████████████████████| 26/26 [07:52<00:00, 18.19s/it]

Processing final_summary with shape: (2048, 4)
Error processing file final_summary.pkl: 'Bit (-1)'
Final ranked results saved successfully!


## Loading of Saved Calculations For HI-VLI

In [8]:
import glob

# Specify the directory where the chunk files are stored
directory_HI_VLI = r"D:\Road to PhD\Data Base\ranked_frequency_results"
summary_HI_VLI = glob.glob(f"{directory_HI_VLI}/*.pkl")

# Load all chunks into a dictionary
summary_HI_VLI_dict = {}
for file in summary_HI_VLI:
    file_name = file.split("\\")[-1]  # Extract the filename (e.g., chunk_100_200.pkl)
    summary_HI_VLI_data = pd.read_pickle(file)
    summary_HI_VLI_dict[file_name] = summary_HI_VLI_data

# Access a specific chunk by its filename
specific_summary = summary_HI_VLI_dict.get("final_ranked_frequency_plus1.pkl")
if specific_summary is not None:
    print("Specific Summary:")
    print(specific_summary.head())
else:
    print("Chunk file not found in the loaded dictionary!")

# Save individual chunks or perform operations
#for name, data in chunk_dict.items():
   # print(f"Preview of {name}:")
    #print(data.head())

Specific Summary:
     Bit (+1)  Frequency (+1)
597       598          767936
1555     1556          673190
856       857          669664
1989     1990          653994
1154     1155          616200


In [19]:
final_bit_freq_plus1 = summary_HI_VLI_dict.get("final_ranked_frequency_plus1.pkl")
final_bit_freq_minus1 = summary_HI_VLI_dict.get("final_ranked_frequency_minus1.pkl")
final_bit_freq_0 = summary_HI_VLI_dict.get("final_ranked_frequency_0.pkl")
print (f'The Top Bits for +1 Are the Following:{final_bit_freq_plus1.head()}')


The Top Bits for +1 Are the Following:     Bit (+1)  Frequency (+1)
597       598          767936
1555     1556          673190
856       857          669664
1989     1990          653994
1154     1155          616200


In [16]:
final_bit_freq_plus1.iloc[:20]

Bit (+1)  Frequency (+1)
597       598          767936
1555     1556          673190
856       857          669664
1989     1990          653994
1154     1155          616200
1084     1085          611846
0           1          609248
2038     2039          606672
791       792          603936
1082     1083          602408
50         51          595882
1862     1863          593712
5           6          591448
18         19          585936
7           8          576194
427       428          572000
1916     1917          565438
1216     1217          551670
1218     1219          545458
836       837          533484

In [17]:
final_bit_freq_minus1.iloc[:20]

Bit (-1)  Frequency (-1)
791       792          335052
1916     1917          334536
0           1          330252
1082     1083          324800
1154     1155          318150
7           8          313484
1084     1085          311190
545       546          299128
2           3          297882
2038     2039          296439
1989     1990          292644
1862     1863          291942
1218     1219          289990
1555     1556          287000
805       806          277263
18         19          269892
1722     1723          264438
1849     1850          263662
5           6          256861
645       646          256540

In [20]:
final_bit_freq_0.iloc[:20]

Bit (0)  Frequency (0)
72        73        1272056
1148    1149        1269766
1430    1431        1269766
537      538        1267699
1124    1125        1265632
1650    1651        1265126
385      386        1264943
283      284        1263565
1279    1280        1262876
1999    2000        1262621
1626    1627        1262432
959      960        1262432
832      833        1262432
1749    1750        1262187
1770    1771        1262187
1163    1164        1262187
58        59        1261980
934      935        1261498
814      815        1261295
41        42        1261058

Notable Results: 
1. bit 792 both have high +1 and -1 values, suggesting that altering its position may result to lesser bioactivity, or maybe synergystic effect, meaning that inorder for it to be active it must have a specific bit beside it (neighboring)
2. bit 1917 has the same case with 792
3. bit 1083 has the same case with 792
4. bit 1 has the same case with 792
5. 1085 has the same case with 792
6. 1990, 1863, 1219, 1556.
7. Bit Frequency 0, these are the bits that is common between the neutral point compounds and high value compounds, therefore, from here you can simply connect them then add the positive bits to generate a bioactive molecule against HCT-116. 

## MI - VLI Counting of Bits 

In [25]:
file_path = r"D:\Road to PhD\Data Base\Simplfied chunks_AC\AC_chunk_AC_0_100_simplified.pkl"
try:
    specific_chunk = pd.read_pickle(file_path)
    print(specific_chunk.info())  # Structure of the DataFrame
    print(specific_chunk.head())  # Preview the data
except Exception as e:
    print(f"Error loading file: {e}")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49382 entries, 0 to 49381
Columns: 2048 entries, 1 to Molecule_ID
dtypes: int32(2047), object(1)
memory usage: 386.0+ MB
None
   1  2  3  4  5  6  7  8  9  10  ...  2039  2040  2041  2042  2043  2044  \
0 -1 -1 -1  0 -1  0  0 -1 -1   0  ...     0    -1    -1     0     0     1   
1 -1 -1 -1  0 -1  0  0 -1 -1   0  ...     0    -1    -1     0     0     1   
2 -1 -1 -1  0 -1  0  0 -1  0   0  ...     1    -1     0     0     0     0   
3 -1 -1 -1  0 -1  0  0 -1  0   0  ...     1    -1     0     0     0     0   
4 -1 -1 -1  0 -1  0  0  0 -1   0  ...     0    -1    -1     0     0     0   

   2045  2046  2047                                        Molecule_ID  
0     0     0     0  ((14, 242), (14, 640), (14, 1926), (14, 2052),...  
1     1     0     0                                      ((76, 5656),)  
2     0     0     0                                      ((93, 5871),)  
3     1     0     0             ((16, 3285), (143, 3285), (144, 3285)

In [26]:
specific_chunk.head()

1  2  3  4  5  6  7  8  9  10  ...  2039  2040  2041  2042  2043  2044  \
0 -1 -1 -1  0 -1  0  0 -1 -1   0  ...     0    -1    -1     0     0     1   
1 -1 -1 -1  0 -1  0  0 -1 -1   0  ...     0    -1    -1     0     0     1   
2 -1 -1 -1  0 -1  0  0 -1  0   0  ...     1    -1     0     0     0     0   
3 -1 -1 -1  0 -1  0  0 -1  0   0  ...     1    -1     0     0     0     0   
4 -1 -1 -1  0 -1  0  0  0 -1   0  ...     0    -1    -1     0     0     0   

   2045  2046  2047                                        Molecule_ID  
0     0     0     0  ((14, 242), (14, 640), (14, 1926), (14, 2052),...  
1     1     0     0                                      ((76, 5656),)  
2     0     0     0                                      ((93, 5871),)  
3     1     0     0             ((16, 3285), (143, 3285), (144, 3285))  
4     0     0     0                           ((71, 2509), (71, 3453))  

[5 rows x 2048 columns]

### Counting and Ranking Bits 


In [67]:
import os
import pandas as pd
from tqdm import tqdm  # For progress tracking

# Directory to save ranked frequency results
directory = r"D:\Road to PhD\Data Base\Simplified chunks_AE"
output_directory = r"D:\Road to PhD\Data Base\ranked_frequency_results_AE"
os.makedirs(output_directory, exist_ok=True)  # Ensure the directory exists

# Initialize empty DataFrames for each frequency category
final_plus1 = pd.DataFrame()
final_minus1 = pd.DataFrame()
final_0 = pd.DataFrame()

# **Fix**: Loop through actual files in the directory instead of `directory.items()`
for file_name in tqdm(os.listdir(directory), desc="Processing simplified chunks"):
    if not file_name.endswith(".pkl"):  # Ensure we process only `.pkl` files
        continue

    chunk_path = os.path.join(directory, file_name)  # Full path to file
    chunk_name = file_name.replace(".pkl", "")  # Remove `.pkl` for cleaner naming

    # Load the chunk DataFrame from the pickle file
    grouped_data = pd.read_pickle(chunk_path)
    print(f"Processing chunk: {chunk_name} with shape {grouped_data.shape}")

    # Step 1: Aggregate frequencies **without storing full Molecule_ID lists**
    plus1_summary = grouped_data.drop(columns=["Molecule_ID"]).apply(lambda x: (x == 1).sum()).reset_index()
    plus1_summary.columns = ['Bit (+1)', 'Frequency (+1)']
    plus1_summary['Chunk_Name'] = chunk_name  # Keep track of source chunk

    minus1_summary = grouped_data.drop(columns=["Molecule_ID"]).apply(lambda x: (x == -1).sum()).reset_index()
    minus1_summary.columns = ['Bit (-1)', 'Frequency (-1)']
    minus1_summary['Chunk_Name'] = chunk_name  

    zero_summary = grouped_data.drop(columns=["Molecule_ID"]).apply(lambda x: (x == 0).sum()).reset_index()
    zero_summary.columns = ['Bit (0)', 'Frequency (0)']
    zero_summary['Chunk_Name'] = chunk_name  

    # Step 2: **Aggregate frequencies across all chunks**
    final_plus1_AE = pd.concat([final_plus1, plus1_summary], ignore_index=True)
    final_minus1_AE = pd.concat([final_minus1, minus1_summary], ignore_index=True)
    final_0_AE = pd.concat([final_0, zero_summary], ignore_index=True)

# Step 3: Rank the frequencies by sum
final_plus1_AE = final_plus1_AE.groupby('Bit (+1)').agg({'Frequency (+1)': 'sum'}).reset_index().sort_values(by='Frequency (+1)', ascending=False)
final_minus1_AE = final_minus1_AE.groupby('Bit (-1)').agg({'Frequency (-1)': 'sum'}).reset_index().sort_values(by='Frequency (-1)', ascending=False)
final_0_AE = final_0_AE.groupby('Bit (0)').agg({'Frequency (0)': 'sum'}).reset_index().sort_values(by='Frequency (0)', ascending=False)

# Step 4: Save ranked results directly **without saving intermediate bit frequency files**
final_plus1_AE.to_pickle(os.path.join(output_directory, "final_ranked_frequency_AE_plus1.pkl"))
final_minus1_AE.to_pickle(os.path.join(output_directory, "final_ranked_frequency_AE_minus1.pkl"))
final_0_AE.to_pickle(os.path.join(output_directory, "final_ranked_frequency_AE_0.pkl"))

print("Final ranked results saved successfully!")

Processing simplified chunks:   0%|                                                             | 0/25 [00:00<?, ?it/s]

Processing chunk: chunk_AE_0_100_simplified with shape (6225, 2048)


Processing simplified chunks:   4%|██                                                   | 1/25 [00:01<00:25,  1.07s/it]

Processing chunk: chunk_AE_1000_1100_simplified with shape (6389, 2048)


Processing simplified chunks:   8%|████▏                                                | 2/25 [00:02<00:26,  1.17s/it]

Processing chunk: chunk_AE_100_200_simplified with shape (6225, 2048)


Processing simplified chunks:  12%|██████▎                                              | 3/25 [00:03<00:23,  1.09s/it]

Processing chunk: chunk_AE_1100_1200_simplified with shape (6059, 2048)


Processing simplified chunks:  16%|████████▍                                            | 4/25 [00:04<00:21,  1.03s/it]

Processing chunk: chunk_AE_1200_1300_simplified with shape (6639, 2048)


Processing simplified chunks:  20%|██████████▌                                          | 5/25 [00:05<00:20,  1.04s/it]

Processing chunk: chunk_AE_1300_1400_simplified with shape (6473, 2048)


Processing simplified chunks:  24%|████████████▋                                        | 6/25 [00:06<00:21,  1.12s/it]

Processing chunk: chunk_AE_1400_1500_simplified with shape (7302, 2048)


Processing simplified chunks:  28%|██████████████▊                                      | 7/25 [00:07<00:19,  1.08s/it]

Processing chunk: chunk_AE_1500_1600_simplified with shape (6225, 2048)


Processing simplified chunks:  32%|████████████████▉                                    | 8/25 [00:08<00:17,  1.04s/it]

Processing chunk: chunk_AE_1600_1700_simplified with shape (5478, 2048)


Processing simplified chunks:  36%|███████████████████                                  | 9/25 [00:09<00:16,  1.05s/it]

Processing chunk: chunk_AE_1700_1800_simplified with shape (6307, 2048)


Processing simplified chunks:  40%|████████████████████▊                               | 10/25 [00:10<00:15,  1.02s/it]

Processing chunk: chunk_AE_1800_1900_simplified with shape (6306, 2048)


Processing simplified chunks:  44%|██████████████████████▉                             | 11/25 [00:11<00:13,  1.01it/s]

Processing chunk: chunk_AE_1900_2000_simplified with shape (6142, 2048)


Processing simplified chunks:  48%|████████████████████████▉                           | 12/25 [00:12<00:13,  1.04s/it]

Processing chunk: chunk_AE_2000_2100_simplified with shape (5892, 2048)


Processing simplified chunks:  52%|███████████████████████████                         | 13/25 [00:13<00:11,  1.00it/s]

Processing chunk: chunk_AE_200_300_simplified with shape (6390, 2048)


Processing simplified chunks:  56%|█████████████████████████████                       | 14/25 [00:14<00:11,  1.03s/it]

Processing chunk: chunk_AE_2100_2200_simplified with shape (5561, 2048)


Processing simplified chunks:  60%|███████████████████████████████▏                    | 15/25 [00:15<00:10,  1.09s/it]

Processing chunk: chunk_AE_2200_2300_simplified with shape (5893, 2048)


Processing simplified chunks:  64%|█████████████████████████████████▎                  | 16/25 [00:16<00:09,  1.08s/it]

Processing chunk: chunk_AE_2300_2400_simplified with shape (5146, 2048)


Processing simplified chunks:  68%|███████████████████████████████████▎                | 17/25 [00:17<00:08,  1.06s/it]

Processing chunk: chunk_AE_2400_2417_simplified with shape (913, 2048)


Processing simplified chunks:  72%|█████████████████████████████████████▍              | 18/25 [00:18<00:07,  1.04s/it]

Processing chunk: chunk_AE_300_400_simplified with shape (7052, 2048)


Processing simplified chunks:  76%|███████████████████████████████████████▌            | 19/25 [00:19<00:06,  1.05s/it]

Processing chunk: chunk_AE_400_500_simplified with shape (6723, 2048)


Processing simplified chunks:  80%|█████████████████████████████████████████▌          | 20/25 [00:21<00:05,  1.05s/it]

Processing chunk: chunk_AE_500_600_simplified with shape (6556, 2048)


Processing simplified chunks:  84%|███████████████████████████████████████████▋        | 21/25 [00:22<00:04,  1.11s/it]

Processing chunk: chunk_AE_600_700_simplified with shape (6972, 2048)


Processing simplified chunks:  88%|█████████████████████████████████████████████▊      | 22/25 [00:23<00:03,  1.09s/it]

Processing chunk: chunk_AE_700_800_simplified with shape (7136, 2048)


Processing simplified chunks:  92%|███████████████████████████████████████████████▊    | 23/25 [00:24<00:02,  1.08s/it]

Processing chunk: chunk_AE_800_900_simplified with shape (6140, 2048)


Processing simplified chunks:  96%|█████████████████████████████████████████████████▉  | 24/25 [00:25<00:01,  1.16s/it]

Processing chunk: chunk_AE_900_1000_simplified with shape (6557, 2048)


Processing simplified chunks: 100%|████████████████████████████████████████████████████| 25/25 [00:26<00:00,  1.07s/it]

Final ranked results saved successfully!


In [68]:
final_plus1_AE.iloc[:20]

Bit (+1)  Frequency (+1)
856        857            2916
838        839            1972
1154      1155            1924
1216      1217            1922
1007      1008            1768
1989      1990            1764
5            6            1710
804        805            1702
1730      1731            1675
834        835            1650
484        485            1650
791        792            1638
231        232            1632
1082      1083            1617
1722      1723            1612
1084      1085            1610
427        428            1600
1555      1556            1600
1862      1863            1575
2            3            1537

In [63]:
final_plus1_AD.iloc[:20]

Bit (+1)  Frequency (+1)
1555      1556           15600
0            1           14993
597        598           14896
7            8           14094
1154      1155           13616
791        792           13338
5            6           12720
427        428           12450
1084      1085           12144
1722      1723           11966
805        806           11715
2            3           11660
2038      2039           11502
690        691           11457
1849      1850           11395
2043      2044           11352
1372      1373           11346
856        857           11340
1989      1990           11268
1082      1083           10890

In [64]:
final_plus1_AC.iloc[:20]


Bit (+1)  Frequency (+1)
1555      1556           15920
597        598           14782
1154      1155           14615
7            8           14526
5            6           13560
0            1           13536
856        857           13014
1722      1723           12834
427        428           12725
2043      2044           12342
1372      1373           12338
2038      2039           12042
2            3           11766
1914      1915           11418
1916      1917           11382
1084      1085           11362
791        792           11271
50          51           10965
18          19           10962
1862      1863           10530

In [58]:
final_plus1_AD.iloc[:100]

Bit (+1)  Frequency (+1)
1555      1556           15920
597        598           14782
1154      1155           14615
7            8           14526
5            6           13560
...        ...             ...
124        125            4752
322        323            4736
1431      1432            4712
655        656            4692
1098      1099            4690

[100 rows x 2 columns]

In [60]:
final_minus1_AC.iloc[:20]

Bit (-1)  Frequency (-1)
1989      1990           16168
1082      1083           16100
1218      1219           15694
645        646           14883
791        792           14800
1862      1863           14450
1916      1917           14356
1084      1085           13596
223        224           12840
50          51           12432
1849      1850           12012
0            1           11872
805        806           11424
2            3           11362
1154      1155           11088
597        598           11070
545        546           10983
2038      2039           10900
2040      2041           10792
803        804           10700

In [33]:
final_0_AC.iloc[:20]

Bit (0)  Frequency (0)
1430     1431          51928
1224     1225          51928
72         73          51849
973       974          51849
1163     1164          51849
157       158          51849
1262     1263          51849
1891     1892          51770
369       370          51770
1650     1651          51770
1092     1093          51770
755       756          51770
210       211          51770
1114     1115          51770
1659     1660          51770
883       884          51770
1118     1119          51691
1520     1521          51691
607       608          51691
537       538          51691